File: C# Examples/14 - 15 Creating Add-ins/MyMacroAddIn - Part2/MyMacroAddIn/CustomProps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SolidWorks.Interop.sldworks;

namespace MyMacroAddIn
{
    class CustomProps
    {
        public bool PropExists(ModelDoc2 swDoc,
            string propName, string config = "")
        {
            CustomPropertyManager cpm;
            cpm = swDoc.Extension.CustomPropertyManager[config];
            string val = cpm.Get(propName);
            if (val == "")
            {
                return false;
            }
            return true;
        }
    }
}

File: 14 - 15 Creating Add-ins/MyMacroAddIn/MyMacroAddIn/CustomProps.vb
Imports SolidWorks.Interop.sldworks
Public Class CustomProps
    Public Function PropExists(swDoc As ModelDoc2, propName As String, Optional config As String = "") As Boolean
        Dim cpm As CustomPropertyManager
        cpm = swDoc.Extension.CustomPropertyManager(config)
        Dim val As String
        val = cpm.Get(propName)
        If val = "" Then
            Return False
        End If
        Return True
    End Function
End Class
